### DISCUSSION TITLE
make all workspaces phantom workspaces

### DISCUSSION THREAD

**Discussion by dezren39** [nikitabobko/AeroSpace#996] [feature-ideas] [answered]
hello,

for reference, on linux i use waybar and hyprland. reading in the issues here I found that there is a concept in aerospace called 'phantom workspace' which is what is used i think for secondary monitors making a workspace 11. what i want is to be able to make all workspaces phantom workpaces. when my pc boots, i'm in workspace 1 and the list in the toolbar only has workspace 1. when i change to workspace 2 it dynamically creates workspace 2. if workspace 1 had no windows, it would automatically remove workspace 1. if i open a window in workspace 2 and then go to workspace 5, then the menu at the top would list workspace 2 and 5. if i hit previous workspace, i would go to 2. if i move out of workspace 5 and no windows open in workspace 5, then workspace 5 would automatically delete. at that point next/prev would loop on the single workspace 2 until there is more than one workspace again.

is there a way to turn on phantom workspace for all workspaces? (or even all workspaces except 1-4?) I don't want to uncomment alt+(1 through 10), i like all that. i just don't want aerospace looking into my config, finding out i _could_ open workspace 9 and then automatically making it for me.

**Comment by danscheer** [comment]
Same Problem for me, how I understand it, workspaces can only be created with a dedicated keybind (alt + 1), its not possible to create a workspace with "prev | next" (to just "open" a workspace next to the current one) and also not possible to move an node to a non-existing workspace. So working very dynamicly with aerospace is just not possible. Am i right here?

**Comment by nikitabobko** [comment]
https://github.com/nikitabobko/AeroSpace/discussions/875#discussioncomment-11618465

**Comment by nikitabobko** [comment, marked as answer]
https://github.com/nikitabobko/AeroSpace/issues/999
